docs/rfcs/0003-sql-datastore.md

```diff
@@ -0,0 +1,199 @@
+- Start Date: 2020-02-22
+- Target Version: 1.0.0
+
+# Summary
+
+This RFC proposes adding SQL database as datastore for PipeCD Control-plane. Currently, we're considering between PostgreSQL and MySQL for those NoSQL support features.
+
+# Motivation
+
+The background of this proposal is we are going to add complex queries and more indexes to the control-plane datastore for some new features of PipeCD, the current in-use NoSQL datastore is blocking those requirements.
+
+# Detailed design
+
+As mention is the motivation section, the root cause of this change is: we're going to add complex queries with those indexes to the datastore of PipeCD control-plane. The use-case of those queries requires:
+- easy way to pagination query result
+- easy way to ordering query result
+- easy way to add indexes for future adding queries
+
+Besides, to keep `simplicity on installing` characteristic of PipeCD, the chosen must be:
+- easy to be installed on cloud provider environment (GCP, AWS, etc)
+- can use the updated version easily
+
+We're using NoSQL for now due to its schemaless characteristic is good for the deployment process, so we focus on NoSQL support features on the chosen SQL database.
+
```

Comment:
How about adding something like this to emphasize the method we use?

`In order to accomplish the schema flexibility on SQL database, we use a JSON field to store our model data containing fluctuant structure fields, just some stable fields like ID, CreatedAt, UpdatedAt will be explicitly declared while defining the table schema.`

Reply:
Nice, I will use your suggestion 👍